rosidl_generator_dds_idl/rosidl_generator_dds_idl/__init__.py

```diff
@@ -52,11 +55,76 @@ def generate_dds_idl(
                     },
                     globals=data,
                 )
-                interpreter.file(open(template_file))
+                interpreter.file(open(template_file_msg))
                 interpreter.shutdown()
             except Exception:
                 os.remove(generated_file)
                 raise
+        elif extension == '.srv':
+            srv_spec = parse_service_file(pkg_name, idl_file)
+            default_value_string = None
+            request_fields = [
+                Field(
+                    Type('int64', context_package_name=pkg_name),
+                    'sequence_id', default_value_string),
+                Field(
+                    Type('%sRequest' % srv_spec.srv_name, context_package_name=pkg_name),
+                    'request', default_value_string)
+                ]
+            response_fields = [
+                Field(
+                    Type('int64', context_package_name=pkg_name),
+                    'sequence_id', default_value_string),
+                Field(
+                    Type('%sResponse' % srv_spec.srv_name, context_package_name=pkg_name),
+                    'response', default_value_string)
+                ]
+            constants = []
+            sample_spec_request = MessageSpecification(
+                srv_spec.pkg_name, 'Sample%sRequest' % srv_spec.srv_name, request_fields, constants)
+            write_sample_spec_request = MessageSpecification(
+                srv_spec.pkg_name, 'WriteSample%sRequest' % srv_spec.srv_name, request_fields,
+                constants)
+
+            sample_spec_response = MessageSpecification(
+                srv_spec.pkg_name, 'Sample%sResponse' % srv_spec.srv_name, response_fields, constants)
+            write_sample_spec_response = MessageSpecification(
+                srv_spec.pkg_name, 'WriteSample%sResponse' % srv_spec.srv_name, response_fields,
+                constants)
+
+            generated_files = [
+                (sample_spec_request,
+                 os.path.join(output_dir, '%s_.idl' % sample_spec_request.base_type.type)),
+                (sample_spec_response,
+                 os.path.join(output_dir, '%s_.idl' % sample_spec_response.base_type.type)),
+                (write_sample_spec_request,
+                 os.path.join(output_dir, '%s_.idl' % write_sample_spec_request.base_type.type)),
+                (write_sample_spec_response,
+                 os.path.join(output_dir, '%s_.idl' % write_sample_spec_response.base_type.type)),
+            ]
+
+            for spec, generated_file in generated_files:
+                print('Generating SERVICE supporting file: %s' % generated_file)
```

Comment:
The message should be symmetric to the message equivalent above. I would also suggest to not use uppercase for it:

    Generating message: %s
    Generating service: %s